src/Engine/Combat/CombatEngine.java

```diff
@@ -25,10 +34,74 @@ public static BattleSummary calculateBattleResults( Unit attacker, Unit defender
+   * @return A BattleSummary object containing all relevant details from this combat instance.
+   */
+  public static BattleSummary calculateBattleResults(Unit attacker, Unit defender, GameMap map, int attackerX, int attackerY, boolean isSim)
+  {
+    int defenderX = defender.x; // This variable is technically not necessary, but provides consistent names with attackerX/Y.
+    int defenderY = defender.y;
+    boolean attackerMoved = attacker.x != attackerX || attacker.y != attackerY;
+    int battleRange = Math.abs(attackerX - defenderX) + Math.abs(attackerY - defenderY);
+    WeaponModel attackerWeapon = attacker.chooseWeapon(defender.model, battleRange, attackerMoved);
+    WeaponModel defenderWeapon = null;
+    // Only attacks at point-blank range can be countered
+    if( 1 == battleRange )
+      defenderWeapon = defender.chooseWeapon(attacker.model, battleRange, false);
+
+    CombatContext context = new CombatContext(map, attacker, attackerWeapon, defender, defenderWeapon, battleRange, attackerX, attackerY);
+    
+    // If the attacker and defender get swapped, we want to still give a coherent BattleSummary.
+    Map<Unit, Entry<WeaponModel,Double>> unitDamageMap = new HashMap<Unit, Entry<WeaponModel,Double>>();
+    unitDamageMap.put(attacker, new AbstractMap.SimpleEntry<WeaponModel,Double>(attackerWeapon, 0.0));
+    unitDamageMap.put(defender, new AbstractMap.SimpleEntry<WeaponModel,Double>(defenderWeapon, 0.0));
 
-    return params.calculateBattleResults();
+    // From here on in, use context variables only
+
+    double attackerHPLoss = 0;
+
+    // Set up our scenario.
+    BattleParams attackInstance = StrikeParams.getAttack(context);
+
+    double defenderHPLoss = attackInstance.calculateDamage();
+    unitDamageMap.put(context.attacker, new AbstractMap.SimpleEntry<WeaponModel,Double>(context.attackerWeapon, defenderHPLoss));
+    if( !isSim && defenderHPLoss > context.defender.getPreciseHP() ) defenderHPLoss = context.defender.getPreciseHP();
+
+    // If the unit can counter, and wasn't killed in the initial volley, calculate return damage.
+    if( context.canCounter && (context.defender.getPreciseHP() > defenderHPLoss) )
+    {
+      // New battle instance with defender counter-attacking.
+      double counterHP = Math.ceil(context.defender.getPreciseHP() - defenderHPLoss); // Account for the first attack's damage to the now-attacker.
+      BattleParams defendInstance = StrikeParams.getCounterAttack(context, counterHP);
+
+      attackerHPLoss = defendInstance.calculateDamage();
+      unitDamageMap.put(context.defender, new AbstractMap.SimpleEntry<WeaponModel,Double>(context.defenderWeapon, attackerHPLoss));
+      if( !isSim && attackerHPLoss > context.attacker.getPreciseHP() ) attackerHPLoss = context.attacker.getPreciseHP();
+    }
+    
+    // Calculations complete. We are setting up our BattleSummary, so go back to using known-accurate info
```

Comment:
Are we using potentially-inaccurate info leading up to this?

Reply:
Depends on what you mean by "accurate".
In my case, I meant that it hasn't been mucked with by the COs so your attacker/defender are based on who started the fight rather than who got firststrike

Reply:
Same here as above. You say just enough here to confuse someone who isn't already intimately familiar with the combat system, and not enough to clarify what you mean.